compiler/daml-extension/src/extension.ts

```diff
@@ -86,6 +94,17 @@ export async function activate(context: vscode.ExtensionContext) {
     context.subscriptions.push(d1, d2, d3, d4, d5);
 }
 
+// Compare the extension version with the one stored in the global state.
+// This will be used to show release notes when the version has changed.
+async function checkVersion(context: ExtensionContext) {
+    let packageFile = path.join(context.extensionPath, 'package.json');
+    let data = fs.readFileSync(packageFile, "utf8");
+    let extensionVersion = JSON.parse(data).version;
+    let recordedVersion = context.globalState.get(versionContextKey);
+    if (!recordedVersion || recordedVersion != extensionVersion ) {
+        context.globalState.update(versionContextKey, extensionVersion);
```

Comment:
Judging from the fact that `update` returns a `Thenable` we should probably wait for this. I /believe/ you can just use `await` on any `Thenable` but take that with a grain of salt.